### PR TITLE
Implement creation of grass paths and farmland

### PR DESCRIPTION
**Pull request by lukeeey**
I still have no idea to add the sound but its not the end of the world

### Changes
I've added the `ItemShovel` and `ItemHoe` interfaces

### CODE REVIEW

File: gomint-api/src/main/java/io/gomint/inventory/item/ItemIronHoe.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
  * @author geNAZt
  * @version 1.0
  */
-public interface ItemIronHoe extends ItemStack {
+public interface ItemIronHoe extends ItemTool {
```
Comment (Janmm14) [inline gomint-api/src/main/java/io/gomint/inventory/item/ItemIronHoe.java]: *ItemHoe

File: gomint-server/src/main/java/io/gomint/server/inventory/item/ItemDiamondHoe.java [hunk 2]
```diff
@@ -25,6 +31,15 @@ public ItemDiamondHoe( short data, int amount, NBTTagCompound nbt ) {
     }
     // CHECKSTYLE:ON
 
+    public boolean interact ( EntityPlayer entity, BlockFace face, Vector clickPosition, Block clickedBlock ) {
+        if( entity instanceof EntityPlayer && (clickedBlock instanceof Dirt || clickedBlock instanceof GrassBlock) ) {
```
Comment (Janmm14) [inline gomint-server/src/main/java/io/gomint/server/inventory/item/ItemDiamondHoe.java]: as entity is already an EntityPlayer instanceof is useless

File: gomint-server/src/main/java/io/gomint/server/world/block/GrassBlock.java [hunk 3]
```diff
@@ -3,6 +3,10 @@
 import io.gomint.inventory.item.ItemDirt;
 import io.gomint.inventory.item.ItemShears;
 import io.gomint.inventory.item.ItemStack;
+import io.gomint.math.Vector;
```
Comment (Janmm14) [inline gomint-server/src/main/java/io/gomint/server/world/block/GrassBlock.java]: Imports added but no code?

**Comment by geNAZt** [comment]
Why don't you use a ItemHoe / ItemShovel implementation class so you don't have to copy the implementation into all sort of types.

Also you don't seem to damage the hoe / shovel? is that correct?

**Review by Janmm14 — COMMENTED** [review]
Woopsie forgot to finish my review.

**Comment by Janmm14** [comment]
You should watch the changes in Grass block fully. There were multiplr imports too much, not just 1 ;)

**Comment by lukeeey** [comment]
Removed. I think this is going to need to be a squash and merge 😆

**Comment by geNAZt** [comment]
Looking good for now. Will be changed in the near future though when i implement the observer pattern for items :)

**Pull request merged**